frontend/src/features/properties/map/MapRouter.tsx

```diff
@@ -0,0 +1,117 @@
+
+interface IMapRouterProps {
+  showSideBar: boolean;
+  setShowSideBar: (show: boolean) => void;
+  onZoom?: (apiProperty?: IPropertyApiModel) => void;
+}
+
+export const MapRouter: React.FunctionComponent<IMapRouterProps> = React.memo(props => {
+  const location = useLocation();
+  const history = useHistory();
+
+  let matched = matchPath(location.pathname, {
+    path: '/mapview/sidebar/*',
+    exact: true,
+    strict: true,
+  });
+
+  React.useEffect(() => {
+    if (matched !== null) {
+      props.setShowSideBar(true);
+    } else {
+      props.setShowSideBar(false);
+    }
+  }, [matched, props]);
+
+  const onClose = () => {
+    history.push('/mapview');
+  };
+
+  const pidQueryString = queryString.parse(location.search).pid?.toString() ?? '';
+  return (
+    <Switch>
+      <AppRoute
+        path={`/mapview/sidebar/research/new`}
+        customRender={() => <AddResearchContainer onClose={onClose} />}
+        claim={Claims.RESEARCH_ADD}
+        exact
+        key={'NewResearch'}
+        title={'Create Research File'}
+      />
+      <AppRoute
+        path={`/mapview/sidebar/research/:researchId`}
+        customRender={({ match }) => (
+          <ResearchContainer researchFileId={Number(match.params.researchId)} onClose={onClose} />
+        )}
+        claim={Claims.RESEARCH_VIEW}
+        exact
+        key={'Research'}
+        title={'Research File'}
+      />
+      <AppRoute
+        path={`/mapview/sidebar/property/:propertyId`}
+        customRender={({ match }) => (
+          <MotiInventoryContainer
+            onClose={onClose}
+            id={Number(match.params.propertyId)}
+            pid={pidQueryString}
+            onZoom={props.onZoom}
+          />
+        )}
+        claim={Claims.PROPERTY_VIEW}
+        exact
+        key={'Property'}
+        title={'Property Information'}
+      />
+      <AppRoute
+        path={`/mapview/sidebar/non-inventory-property/:pid`}
+        customRender={({ match }) => (
+          <MotiInventoryContainer onClose={onClose} pid={match.params.pid} onZoom={props.onZoom} />
+        )}
+        claim={Claims.PROPERTY_VIEW}
+        exact
+        key={'PropertyNonInventory'}
+        title={'Property Information - Non Inventory'}
+      />
+      <AppRoute
+        path={`/mapview/sidebar/acquisition/new`}
+        customRender={() => <AddAcquisitionContainer onClose={onClose} />}
+        claim={Claims.ACQUISITION_ADD}
+        exact
+        key={'NewAquisition'}
+        title={'Create Acquisition File'}
+      />
+      <AppRoute
+        path={`/mapview/sidebar/acquisition/:id`}
+        customRender={({ match }) => (
+          <AcquisitionContainer acquisitionFileId={Number(match.params.id)} onClose={onClose} />
+        )}
+        claim={Claims.ACQUISITION_VIEW}
+        exact
+        key={'Aquisition'}
+        title={'Acquisition File'}
+      />
+      <AppRoute
+        path="*"
+        customRender={({ match }) => <>Well this sucks{match.path}</>}
```

Comment:
We do have a 404 page, lets maybe use that instead as amusing as that is.

Reply:
This is only here for development. Will remove it once I checked the routes work as expected